Miracast/MiracastService/MiracastController.cpp

```diff
@@ -0,0 +1,1968 @@
+{
+    MIRACASTLOG_TRACE("Entering...");
+
+#ifdef ENABLE_MIRACAST_SERVICE_TEST_NOTIFIER
+    destroy_TestNotifier();
+#endif
+    send_msg_thunder_msg_hdler_thread(MIRACAST_SERVICE_SHUTDOWN);
+
+    if (nullptr != m_p2p_ctrl_obj)
+    {
+        MiracastP2P::destroyInstance();
+        m_p2p_ctrl_obj = nullptr;
+    }
+    if (nullptr != m_controller_thread){
+        delete m_controller_thread;
+        m_controller_thread = nullptr;
+    }
+    if (nullptr != m_thunder_req_handler_thread ){
+        delete m_thunder_req_handler_thread;
+        m_thunder_req_handler_thread = nullptr;
+    }
+    MIRACASTLOG_TRACE("Exiting...");
+    return MIRACAST_OK;
+}
+
+std::string MiracastController::parse_p2p_event_data(const char *tmpBuff, const char *lookup_data)
+{
+    char return_buf[1024] = {0};
+    const char  *ret = nullptr, 
+                *ret_equal = nullptr,
+                *ret_space = nullptr,
+                *single_quote_start = nullptr,
+                *single_quote_end = nullptr;
+    ret = strstr(tmpBuff, lookup_data);
+
+    if (nullptr != ret)
+    {
+        if (0 == strncmp(ret, lookup_data, strlen(lookup_data)))
+        {
+            ret_equal = strstr(ret, "=");
+            ret_space = strstr(ret_equal, " ");
+
+            if (0 == strncmp("name", lookup_data, strlen(lookup_data))){
+                single_quote_start = strstr(ret_equal, "'");
+                single_quote_end = strstr(single_quote_start + 1, "'");
+            }
+
+            if (single_quote_start && single_quote_end) {
+                unsigned int length = single_quote_end - single_quote_start;
+                if (length < sizeof(return_buf)) {
+                    snprintf(return_buf, length, "%s", single_quote_start + 1);
+                }
+            }
+            else if (ret_space)
+            {
+                snprintf(return_buf, (int)(ret_space - ret_equal), "%s", ret + strlen(lookup_data) + 1);
+                MIRACASTLOG_VERBOSE("Parsed Data is - %s", return_buf);
+            }
+            else
+            {
+                snprintf(return_buf, strlen(ret_equal - 1), "%s", ret + strlen(lookup_data) + 1);
+                MIRACASTLOG_VERBOSE("Parsed Data is - %s", return_buf);
+            }
+        }
+    }
+    if (return_buf != nullptr)
+        return std::string(return_buf);
+    else
+        return std::string(" ");
+}
+
+std::string MiracastController::start_DHCPClient(std::string interface, std::string &default_gw_ip_addr)
+{
+    MIRACASTLOG_TRACE("Entering...");
+    char command[128] = {0};
+    char sys_cls_file_ifidx[128] = {0};
+    std::string local_addr = "",
+                gw_ip_addr = "",
+                popen_buffer = "";
+    FILE *popen_file_ptr = nullptr;
+    char *current_line_buffer = nullptr;
+    std::size_t len = 0;
+    unsigned char retry_count = 3;
+
+    sprintf( sys_cls_file_ifidx , "/sys/class/net/%s/ifindex" , interface.c_str());
```

Comment:
is this amlogic specifc?

Reply:
No. It is generic linux check to whether iface created or not